src/libcore/slice/mod.rs

```diff
@@ -2466,6 +2466,256 @@ impl<T> [T] {
+    ///
+    /// Returns a new slice pointing to the first `len` elements of the old slice.
+    ///
+    /// Returns `None` if `len > slice.len()`
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(slice_take)]
+    ///
+    /// let mut slice: &mut [_] = &mut ['a', 'b', 'c'];
+    /// let mut first_two = core::slice::take_front_mut(&mut slice, 2).unwrap();
+    /// first_two[1] = 'f';
+    ///
+    /// assert_eq!(slice, &['c']);
+    /// assert_eq!(first_two, &['a', 'f']);
+    /// ```
+    #[inline]
+    #[unstable(feature = "slice_take", issue = "62280")]
+    pub fn take_front_mut<'a>(
+        slice: &mut &'a mut Self,
+        len: usize,
+    ) -> Option<&'a mut Self> {
+        if len > slice.len() {
+            return None;
+        }
+        let original = Self::take_mut(slice);
+        let (front, back) = original.split_at_mut(len);
+        *slice = back;
+        Some(front)
+    }
+
+    /// Takes the last `len` elements out of the slice.
+    ///
+    /// Returns a new slice pointing to the last `len` elements of the old slice.
+    ///
+    /// Returns `None` if `len > slice.len()`
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(slice_take)]
+    ///
+    /// let mut slice: &[_] = &['a', 'b', 'c'];
+    /// let last_two = core::slice::take_back(&mut slice, 2).unwrap();
+    ///
+    /// assert_eq!(slice, &['a']);
+    /// assert_eq!(last_two, &['a', 'b']);
+    /// ```
+    #[inline]
+    #[unstable(feature = "slice_take", issue = "62280")]
+    pub fn take_back<'a>(
+        slice: &mut &'a Self,
+        len: usize,
+    ) -> Option<&'a Self> {
+        if len > slice.len() {
+            return None;
+        }
+        let original = Self::take(slice);
+        let (front, back) = original.split_at(len);
+        *slice = front;
+        Some(back)
+    }
+
+    /// Takes the last `len` elements out of the mutable slice.
+    ///
+    /// Returns a new slice pointing to the last `len` elements of the old slice.
+    ///
+    /// Returns `None` if `len > slice.len()`
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(slice_take)]
+    ///
+    /// let mut slice: &mut [_] = &mut ['a', 'b', 'c'];
+    /// let mut last_two = core::slice::take_back_mut(&mut slice, 2).unwrap();
+    /// last_two[1] = 'f';
+    ///
+    /// assert_eq!(slice, &['a']);
+    /// assert_eq!(last_two, &['b', 'f']);
+    /// ```
+    #[inline]
+    #[unstable(feature = "slice_take", issue = "62280")]
+    pub fn take_back_mut<'a>(
+        slice: &mut &'a mut Self,
+        len: usize,
+    ) -> Option<&'a mut Self> {
+        if len > slice.len() {
+            return None;
+        }
+        let original = Self::take_mut(slice);
+        let (front, back) = original.split_at_mut(len);
+        *slice = front;
+        Some(back)
+    }
+
+    /// Takes the first element out of the slice.
+    ///
+    /// Returns a reference pointing to the first element of the old slice.
```

Comment:
There is no such thing as an old slice, given there is not really a new slice either.